test/unit-tests/config.test.ts

```diff
@@ -80,23 +80,49 @@ suite('Configuration', () => {
         expect(conf.parallelJobs).to.eq(4);
     });
 
-    test('Listen for config changes', () => {
+    test('Listen for config changes', async () => {
         const conf = createConfig({ parallelJobs: 22 });
         let jobs = conf.parallelJobs;
         expect(jobs).to.eq(22);
-        conf.onChange('parallelJobs', j => jobs = j);
```

Comment:
I am trying to understand why do we need a promise here? why the previous approach not working?

Reply:
The previous approach seems to work usually, but there is async going on that isn't being accounted for. I don't think it currently matters, but if it ever runs on another thread, this test will break unless we fix it.